diff -up kservice-5.50.0/src/CMakeLists.txt.0000~ kservice-5.50.0/src/CMakeLists.txt
--- kservice-5.50.0/src/CMakeLists.txt.0000~	2018-09-03 13:45:58.030908859 +0200
+++ kservice-5.50.0/src/CMakeLists.txt	2018-09-03 13:46:30.413484388 +0200
@@ -159,7 +159,7 @@ add_dependencies(KF5Service copy_service
 if (WIN32)
 install( FILES applications.menu  DESTINATION  ${KDE_INSTALL_DATAROOTDIR}/xdg/menus  RENAME ${APPLICATIONS_MENU_NAME} )
 else ()
-install( FILES applications.menu  DESTINATION  ${KDE_INSTALL_SYSCONFDIR}/xdg/menus  RENAME ${APPLICATIONS_MENU_NAME} )
+install( FILES applications.menu  DESTINATION  ${KDE_INSTALL_SYSCONFDIR}/xdg/kde5/menus  RENAME ${APPLICATIONS_MENU_NAME} )
 endif ()
 
 # Local copy for the unittests
diff -up kservice-5.50.0/src/sycoca/vfolder_menu.cpp.0000~ kservice-5.50.0/src/sycoca/vfolder_menu.cpp
--- kservice-5.50.0/src/sycoca/vfolder_menu.cpp.0000~	2018-09-02 21:32:29.000000000 +0200
+++ kservice-5.50.0/src/sycoca/vfolder_menu.cpp	2018-09-03 13:45:58.030908859 +0200
@@ -739,8 +739,13 @@ VFolderMenu::locateMenuFile(const QStrin
     }
 
     QString result;
+    QString xdgMenuPrefix;
+
+    if(qEnvironmentVariableIsSet("XDG_MENU_PREFIX"))
+        xdgMenuPrefix = QString::fromLocal8Bit("kde-");
+    else
+        xdgMenuPrefix = QString::fromLocal8Bit(qgetenv("XDG_MENU_PREFIX"));
 
-    QString xdgMenuPrefix = QString::fromLocal8Bit(qgetenv("XDG_MENU_PREFIX"));
     if (!xdgMenuPrefix.isEmpty()) {
         QFileInfo fileInfo(fileName);
 
