diff -urN kservice-5.9.0/src/kbuildsycoca/CMakeLists.txt kservice-5.9.0-patched/src/kbuildsycoca/CMakeLists.txt
--- kservice-5.9.0/src/kbuildsycoca/CMakeLists.txt	2015-04-04 22:12:17.000000000 +1000
+++ kservice-5.9.0-patched/src/kbuildsycoca/CMakeLists.txt	2015-05-08 17:44:56.636099855 +1000
@@ -32,5 +32,5 @@
 if (WIN32)
 install( FILES applications.menu  DESTINATION  ${KDE_INSTALL_DATAROOTDIR}/xdg/menus )
 else ()
-install( FILES applications.menu  DESTINATION  ${KDE_INSTALL_SYSCONFDIR}/xdg/menus )
+install( FILES applications.menu  DESTINATION  ${KDE_INSTALL_SYSCONFDIR}/xdg/kde5/menus )
 endif ()
diff -urN kservice-5.9.0/src/kbuildsycoca/kbuildsycoca.cpp kservice-5.9.0-patched/src/kbuildsycoca/kbuildsycoca.cpp
--- kservice-5.9.0/src/kbuildsycoca/kbuildsycoca.cpp	2015-04-04 22:12:17.000000000 +1000
+++ kservice-5.9.0-patched/src/kbuildsycoca/kbuildsycoca.cpp	2015-05-08 17:45:38.232101748 +1000
@@ -307,7 +307,7 @@
             g_vfolder->setTrackId(m_trackId);
         }
 
-        VFolderMenu::SubMenu *kdeMenu = g_vfolder->parseMenu(QStringLiteral("applications.menu"));
+        VFolderMenu::SubMenu *kdeMenu = g_vfolder->parseMenu(QStringLiteral("kde-applications.menu"));
 
         KServiceGroup::Ptr entry = g_buildServiceGroupFactory->addNew(QStringLiteral("/"), kdeMenu->directoryFile, KServiceGroup::Ptr(), false);
         entry->setLayoutInfo(kdeMenu->layoutList);
diff -urN kservice-5.9.0/src/kbuildsycoca/vfolder_menu.cpp kservice-5.9.0-patched/src/kbuildsycoca/vfolder_menu.cpp
--- kservice-5.9.0/src/kbuildsycoca/vfolder_menu.cpp	2015-04-04 22:12:17.000000000 +1000
+++ kservice-5.9.0-patched/src/kbuildsycoca/vfolder_menu.cpp	2015-05-08 17:45:38.232101748 +1000
@@ -739,7 +739,7 @@
 
     QString result;
 
-    QString xdgMenuPrefix = QString::fromLocal8Bit(qgetenv("XDG_MENU_PREFIX"));
+    QString xdgMenuPrefix = QString::fromLocal8Bit(qgetenv("kde-"));
     if (!xdgMenuPrefix.isEmpty()) {
         QFileInfo fileInfo(fileName);
 
